src/platforms/stm32f4-discovery.c

```diff
@@ -462,17 +462,17 @@ void dma1_stream3_isr(void) {
                                ? (const uint16_t *)spi_rx_raw_adc[1]
                                : (const uint16_t *)spi_rx_raw_adc[0];
 
-  bool fault = !adc_response_is_valid(sequence);
-
-  for (int i = 0; i < NUM_SENSORS; ++i) {
-    if (config.sensors[i].source == SENSOR_ADC) {
-      config.sensors[i].fault = fault ? FAULT_CONN : FAULT_NONE;
-      config.sensors[i].raw_value =
-        read_adc_pin(sequence, config.sensors[i].pin);
-    }
+  struct adc_update update = {
+    .time = current_time(),
```

Comment:
Likewise, use predicted time.  The sample rate should be very consistent, time values should indicate as such.